### PR TITLE
Minimize lodash size in dependencies

### LINKED ISSUE #15029: Remove lodash.flatmap
Lodash minimal imports like lodash/flatmap to be removed

https://output.circle-artifacts.com/output/job/0b87725e-d9fa-4b1e-87c8-209ba5bb939a/artifacts/0/build-artifacts/source-map-explorer/common-6.html

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

In order to optimize the size taken by lodash library, all `lodash.*` packages usage should be avoided, preferring the usage of the main `lodash` package of the same version across all dependencies.

* `lodash.flatmap` removed

**Note**
`lodash.throttle` and `lodash.isequalwith` ignored as they will soon be removed along with `3box`

## Changes

* `eth-json-rpc-filters` has been updated to `^5.0.0` _(v5 doesn't use `lodash.flatmap`)_
* Forced resolution for `**/web3-provider-engine/eth-json-rpc-filters` to `^5.0.0`  _(as `web3-provider-engine` no longer accepts PRs)_

## More Information

* Fixes #15029

### CODE REVIEW

**Comment by mikesposito** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by segun** [comment]
This PR compiles and runs successfully, but I believe proper QA (@seaona) will be needed.
Also someone else who knows lavamoat should be requested to review (@danjm)

**Comment by brad-decker** [comment]
@mikesposito there's some changes to the lavamoat files that don't make sense to me, could you do me a favor and blowout your node_modules with `rm -rf node_modules` and rerun yarn setup and yarn lavamoat:auto?

**Review by brad-decker — DISMISSED** [review]
Upon closer review, I realize that the changes I wasn't understanding are due to the resolution not upgrade of eth-json-rpc-filters. Essentially async-mutex is hoisted into metamask/controllers>eth-json-rpc-infura due to alignment between those packages now. This also adds a new top level entry for eth-json-rpc-filters>async-mutex

**Review by brad-decker — DISMISSED** [review]

**Comment by adonesky1** [comment]
@mikesposito looks like this needs a rebase! Sorry for letting it sit for so long!

File: lavamoat/build-system/policy.json [hunk 1]
```diff
@@ -957,6 +957,70 @@
         "gulp>gulp-cli>isobject": true
       }
     },
+    "@lavamoat/allow-scripts>@npmcli/run-script>node-gyp>npmlog": {
```
Comment (adonesky1) [inline lavamoat/build-system/policy.json:960]: not sure we want to add the diff of this file! I believe it's platform dependent. Is that right @kumavis @brad-decker @legobeat?

File: package.json [hunk 2]
```diff
@@ -261,7 +262,7 @@
     "end-of-stream": "^1.4.4",
     "eth-block-tracker": "^6.0.0",
     "eth-ens-namehash": "^2.0.8",
-    "eth-json-rpc-filters": "^4.2.1",
+    "eth-json-rpc-filters": "^5.0.0",
```
Comment (adonesky1) [inline package.json:265]: I confirmed the only breaking change from v4.2.1 to 5 is minimum node version bump to 14.

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]

**Pull request merged**